### PR TITLE
Add Thingy91 support

### PR DESCRIPTION
**Pull request by bjda**
Enables mcuboot to work on the new Nordic Thingy:91.

This PR is a superset of #44 , which should be closed.

### CODE REVIEW

**Comment by bjda** [comment]
@mbolivar @nvlsianpu @pdunaj ping

File: boot/zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -54,6 +54,9 @@ macro(app_set_runner_args)
   endif()
 endmacro()
 
+# nRF Connect SDK application boilerplate, needed for out-of-tree Nordic boards.
```
Comment (nvlsianpu) [inline boot/zephyr/CMakeLists.txt]: @SebastianBoe Is this line OK.  Can this interfere with boilerplate bellow?

File: boot/zephyr/Kconfig [hunk 2]
```diff
@@ -174,7 +174,7 @@ config BOOT_MAX_IMG_SECTORS
 
 config BOOT_ERASE_PROGRESSIVELY
 	bool "Erase flash progressively when receiving new firmware"
-	default y if SOC_NRF52840
+	default y if SOC_NRF52840 || SOC_NRF9160
```
Comment (nvlsianpu) [inline boot/zephyr/Kconfig]: What the reason for enabling this for all nRF9160?

File: boot/zephyr/CMakeLists.txt (hunk 1 above)
Comment (SebastianBoe) [reply boot/zephyr/CMakeLists.txt]: This is OK (and best-practice for NCS application's).

File: boot/zephyr/Kconfig (hunk 2 above)
Comment (bjda) [reply boot/zephyr/Kconfig]: This avoids a ~10 second delay at the beginning of the flashing process. Given that is saves this much time in a typical use case, I considered it a sensible default.

**Comment by mbolivar** [comment]
These seem fine to me, but shouldn't the `[nrf toup] zephyr: add support for Thingy:91` patch be `noup`, not `toup`? That board definition is downstream only, so I don't think upstream mcuboot would want that patch.

@nvlsianpu, what do you think?

**Comment by bjda** [comment]
@mbolivar I was wondering about that. I expect the board itself will be upstreamed quite soon (it is public after all). If we should wait until after the fact, say the word and i will change it to `[noup]`.

**Comment by mbolivar** [comment]
Hi @bjda -- I think it would be cleaner to keep this noup until the decision is definitely clear we will upstream the board. If that decision has been made, never mind me :).

**Comment by joakimtoe** [comment]
@mbolivar : Yes, the board is to be upstreamed.

**Comment by mbolivar** [comment]
> @mbolivar : Yes, the board is to be upstreamed.

In that case this looks fine to me as a temporary measure until we have time to upstream it out of the nrf repo. Thanks!

**Pull request merged**